group/short.go

```diff
@@ -132,13 +140,19 @@ func (e *wElt) MarshalBinaryCompress() ([]byte, error) {
 func (e *wElt) UnmarshalBinary(b []byte) error {
 	byteLen := (e.Params().BitSize + 7) / 8
 	l := len(b)
-	switch {
-	case l == 1 && b[0] == 0x00: // point at infinity
+	if l < 1 {
+		return ErrUnmarshal
+	}
+	switch b[0] {
+	case 0x00: // point at infinity
```

Comment:
Should there be a check here for `l == 1`?

Reply:
done